### PR TITLE
feat(komga): filter history and updates by active server

### PR DESCRIPTION
**Pull request by Mister-album**
## Summary

- filter History and Updates by the currently active Komga server profile
- immediately resubscribe when the active server changes and show an empty list when no server is active
- pass an optional `sourceId` through the domain, repository, and SQLDelight query layers while preserving the existing all-sources behavior for null callers
- keep the Updates database subscription stable when only the downloaded-state filter changes
- add domain tests for explicit source filtering and null compatibility

This is a manual port of the useful history/updates portion from the old `epub-split-unrelated-leftovers-2026-07-14` stash. The old stash was not applied directly.

## Validation

- user manual testing passed
- `./gradlew.bat spotlessApply --no-daemon`
- `./gradlew.bat spotlessCheck :data:generateDebugDatabaseInterface :domain:testDebugUnitTest :app:compileDebugKotlin --no-daemon`

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/updates/UpdatesScreenModel.kt [hunk 1]
```diff
@@ -80,25 +83,32 @@ class UpdatesScreenModel(
...
+                }
+
+            combine(
+                databaseUpdates,
                 downloadCache.changes,
                 downloadManager.queueState,
-                // needed for Kotlin filters (downloaded)
                 getUpdatesItemPreferenceFlow().distinctUntilChanged { old, new ->
                     old.filterDownloaded == new.filterDownloaded
                 },
```
[the remark this answers is not in the document]
Comment (Mister-album) [reply app/src/main/java/eu/kanade/tachiyomi/ui/updates/UpdatesScreenModel.kt]: Fixed in 047e8b930. The outer combine now maps the preference flow directly to filterDownloaded and applyDownloadedFilter accepts only that TriState, so no stale unrelated preference fields can leak into future Kotlin-side filtering. Spotless, SQLDelight generation, domain tests, and app Kotlin compilation pass.

File: app/src/main/java/eu/kanade/tachiyomi/ui/history/HistoryScreenModel.kt [hunk 2]
```diff
@@ -71,10 +75,17 @@ class HistoryScreenModel(
...
-                    getHistory.subscribe(query ?: "")
+            combine(
+                state.map { it.searchQuery }.distinctUntilChanged(),
+                komgaServerPreferences.activeServerId.changes().distinctUntilChanged(),
+            ) { query, sourceId -> query to sourceId }
+                .flatMapLatest { (query, sourceId) ->
+                    val history = if (sourceId == KomgaServerPreferences.NO_ACTIVE_SERVER) {
+                        flowOf(emptyList())
+                    } else {
+                        getHistory.subscribe(query ?: "", sourceId)
```
[the remark this answers is not in the document]
Comment (Mister-album) [reply app/src/main/java/eu/kanade/tachiyomi/ui/history/HistoryScreenModel.kt:86]: Fixed in 047e8b930. The History clear action now captures the active sourceId and threads it through RemoveHistory and HistoryRepository to a source-scoped SQL delete. NO_ACTIVE_SERVER is a no-op, while the nullable domain default preserves the explicit global-delete API for other callers. Added RemoveHistory sourceId/default compatibility tests; full local validation passes.
